modules/collection/programs/kitty.nix

```diff
@@ -0,0 +1,120 @@
+      description = ''
+        [kitty.conf's online reference]: https://sw.kovidgoyal.net/kitty/conf/
+
+        The configuration converted and written to {file}`$HOME/.config/kitty/kitty.conf`.
+        Please reference [kitty.conf's online reference] or the {manpage}`kitty.conf(5)` for config options.
+      '';
+    };
+
+    theme = {
+      default = mkOption {
+        type = nullOr path;
+        default = null;
+        example = "${pkgs.kitty-themes}/share/kitty-themes/themes/Modus_Vivendi.conf";
+        description = ''
+          [kitty-themes' repository]: https://github.com/kovidgoyal/kitty-themes
+
+          Single theme to be linked to {file}`$HOME/.config/kitty/current-theme.conf`.
+          Please reference [kitty-themes' repository] for available themes.
+        '';
+      };
+      light = mkOption {
+        type = nullOr path;
+        default = null;
+        example = "${pkgs.kitty-themes}/share/kitty-themes/themes/Modus_Operandi.conf";
+        description = ''
+          [kitty-themes' repository]: https://github.com/kovidgoyal/kitty-themes
+
+          Light theme to be linked to {file}`$HOME/.config/kitty/light-theme.auto.conf`.
+          Please reference [kitty-themes' repository] for available themes.
+        '';
+      };
+      dark = mkOption {
+        type = nullOr path;
+        default = null;
+        example = "${pkgs.kitty-themes}/share/kitty-themes/themes/Modus_Vivendi.conf";
+        description = ''
+          [kitty-themes' repository]: https://github.com/kovidgoyal/kitty-themes
+
+          Dark theme to be linked to {file}`$HOME/.config/kitty/dark-theme.auto.conf`.
+          Please reference [kitty-themes' repository] for available themes.
+        '';
+      };
+    };
+
+    extraConfFiles = mkOption {
+      type = attrsOf str;
+      default = {};
+      example = {
+        "diff.conf" = ''
+          num_context_lines 3
+          ignore_name .git
+        '';
+      };
+      description = ''
+        [kitty kittens' online reference]: https://sw.kovidgoyal.net/kitty/kittens_intro/
+
+        Extra configuration files written to {file}`$HOME/.config/kitty/`.
+        Please reference [kitty kittens' online reference] for options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    packages = [cfg.package];
+    files =
+      {
+        ".config/kitty/kitty.conf".source = mkIf (cfg.settings != {}) (
+          keyValue.generate "kitty.conf" cfg.settings
+        );
+
+        ".config/kitty/current-theme.conf".source = mkIf (cfg.theme.default != "") cfg.theme.default;
+        ".config/kitty/light-theme.auto.conf".source = mkIf (cfg.theme.light != "") cfg.theme.light;
+        ".config/kitty/dark-theme.auto.conf".source = mkIf (cfg.theme.dark != "") cfg.theme.dark;
+      }
+      // optionalAttrs (cfg.extraConfFiles != {}) (mapAttrs'
+        (
+          name: val:
+            nameValuePair ".config/kitty/${name}" {text = val;}
+        )
+        cfg.extraConfFiles);
```

Comment:
I'd prefer having `include` statements be generated for each file and *not* symlink them into the kitty config directory.

E.g., I have the following in my config:
```nix
extraConfig = ''
  include ${pkgs.kitty-themes}/share/kitty-themes/themes/moonlight.conf
'';
```
which includes the nix store path to the conf file, no need to clutter $HOME.

Reply:
TBF, now that I think about it, that can simply be done by manually writing down an `include` directive as above, we don't need an `extraConfFiles` at all.

Reply:
I'd be more than okay with an `extraConfig` option, this is a much simpler solution and again, we could give an example for themes here

Reply:
Unsure, I scrolled through the Kitty config docs a good bit, and none of the options really justify having an `extraConfig` either. Even for the include, `settings.include = [];` should suffice.

I could be wrong, feel free to point out if I'm missing something